### ISSUE TITLE
tf.where outputs the wrong tensor dtype

### ISSUE THREAD

**Issue by JustinLiang** [tensorflow/tensorflow#11525]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution**: Ubuntu 14.04
- **TensorFlow installed from**: Source
- **TensorFlow version**: ('v1.2.0-5-g435cdfc', '1.2.1')
- **Python version**: 2.7.13
- **Bazel version**: N/A
- **CUDA/cuDNN version**: Cuda 8.0
- **GPU model and memory**: N/A
- **Exact command to reproduce**:

```
x = tf.Variable([2, 1], dtype=tf.float32)
y = tf.Variable(2, dtype=tf.float32)
z = tf.where(tf.equal(x, y))
print z

>> Tensor("Where_1:0", shape=(?, 1), dtype=int64)
```

### Describe the problem
In the documentation for [tf.where.py](https://www.tensorflow.org/api_docs/python/tf/where), it states that it returns `A Tensor with the same type and shape as x, y if they are non-None.` In the example above, we see that `x` and `y` are both dtype `tf.float32`. Yet, the output is `tf.int64`. Was this intended or a bug?

This bug is a bit problematic for me as I use `tf.where` in my loss function so I need to ensure that all my tensors are `tf.float32` since `tf.cast` is not a differentiable `ops`.

**Comment by printdhruv** [comment]
(1) Please review guidelines [here](https://github.com/tensorflow/tensorflow/blob/master/ISSUE_TEMPLATE.md) before raising an issue 
(2) Following is a definition of tf.where and comments to match with your code.
 ```
tf.where (
    condition,                                 // tf.equal(x,y) , Dimensions must be equal
    x = None,                                  // x = x "Your variable x"
    y = None,                                  // y = y "Your variable y"
    name = None                                // Name = "String"
)
```
(3) Please see the following code for your reference,

```
x = tf.Variable([2, 1], dtype = tf.float32)
y = tf.Variable([2, 3], dtype = tf.float32)
z = tf.where(tf.equal(x,y), x , y , name = "float_Equal")
print(z)
```

Output : -

`Tensor("float_Equal_2:0", shape=(2,), dtype=float32)`
